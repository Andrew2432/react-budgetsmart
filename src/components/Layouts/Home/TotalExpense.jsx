import React, { useContext } from 'react';
import ExpenseContext from '../../../context/Expense/expenseContext';
import { CardPanel } from 'react-materialize';

const TotalExpense = () => {
  const expenseContext = useContext(ExpenseContext);
  const { totalExpense } = expenseContext;

  return (
    <CardPanel
      className="deep-orange darken-1 white-text"
      style={{ textAlign: 'center' }}
    >
      <h4>Total expense</h4>
      <h4>Rs. {totalExpense}</h4>
    </CardPanel>
  );
};

export default TotalExpense;
