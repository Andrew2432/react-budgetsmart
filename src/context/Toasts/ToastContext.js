import { createContext } from 'react';

// Create a context for toasts and export it
const ToastContext = createContext();

export default ToastContext;
