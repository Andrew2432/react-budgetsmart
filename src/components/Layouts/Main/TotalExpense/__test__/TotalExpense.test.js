import TestRenderer from 'react-test-renderer';
import React from 'react';
import TotalExpense from '../TotalExpense';
import ToastContext from '../../../../../context/Toasts/ToastContext';
import ExpenseContext from '../../../../../context/Expense/expenseContext';

describe('AddExpense component', () => {
  test('Renders correctly', () => {
    const totalExpense = new TestRenderer.create(
      (
        <ExpenseContext.Provider value={{ expenses: [] }}>
          <ToastContext.Provider value={{ messages: [] }}>
            <TotalExpense />
          </ToastContext.Provider>
        </ExpenseContext.Provider>
      )
    );
    expect(totalExpense).toMatchSnapshot();
  });
});
