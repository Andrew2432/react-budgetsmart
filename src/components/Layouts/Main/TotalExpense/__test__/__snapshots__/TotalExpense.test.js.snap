// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddExpense component Renders correctly 1`] = `
<div
  className="card-panel deep-orange darken-1"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <h4>
    Total expense
  </h4>
  <h4>
    Rs. 
  </h4>
</div>
`;
